aerogear/aerogear-unifiedpush-server
# KC Decoupling to old 1.7.0

replaces https://github.com/aerogear/aerogear-unifiedpush-server/pull/751  

## How to test
- Configure/run Keycloak
  - Easiest is: run the `keycloak-server.yaml` docker-compose file for Keycloak
  - check the IP address of the keycloak instance (e.g. via `docker inspect`)
  - login to Keycloak (`admin:admin`) and import the _sample_ realm (located in `keycloak/ups-realm-sample.json), which assumes that your WildFly for the UPS is on `localhost` and `8080`
- Run the AG-PUSH WAR file
  - Configure WF 10.1 (if not yet done), like [here](https://aerogear.org/docs/unifiedpush/ups_userguide/index/#server-installation)
  - in your WildFly 10.1, remove the current `auth-server` deployment (if present)
  - start the WF server with the following system properties: `-Dups.realm.name=aerogear -Dups.auth.server.url=http://IP_ADDRESS_OF_KEYCLOAK:8080/auth`
  - copy the newly build `ag-push.war` file to WF's deployment folder

## Expected:
* Login is redirected to the `IP OF KEYCLOAK`
* After successful UPS login, the UI comes up as normal
